### ISSUE TITLE
Use of a number in loop attribute for amp-bodymovin-animation component gives a validation error

### ISSUE THREAD

**Issue by pierlon** [ampproject/amphtml#29890]
**Please only file reports about bugs in AMP here.**

## What's the issue?

When a number is used for the `loop` attribute of the `amp-bodymovin-animation` component, a validation error is shown even though the docs state the use of a number is permitted:

https://github.com/ampproject/amphtml/blob/b6a8ac67874fd912289253d32a3439614d0e79bc/extensions/amp-bodymovin-animation/amp-bodymovin-animation.md#L55

![image](https://user-images.githubusercontent.com/16200219/90586332-791ded00-e1c6-11ea-8852-140c1802612e.png)

## How do we reproduce the issue?

This can be reproduced by using one of the examples in the docs:

https://github.com/ampproject/amphtml/blob/b6a8ac67874fd912289253d32a3439614d0e79bc/extensions/amp-bodymovin-animation/amp-bodymovin-animation.md#L36-L43

## What browsers are affected?

All browsers.

## Which AMP version is affected?

Tested on version 2007302351001.

**Comment by pierlon** [comment]
This seems to be where the issue lies:

https://github.com/ampproject/amphtml/blob/3a86226fe428ce72adb67cffe2dd2f1fae278a35/validator/js/engine/validator.js#L3799-L3804

**Comment by pierlon** [comment]
Also, the spec for the component lists `number` as a valid value, but I suppose the validator should instead be checking if the attribute value is a valid number:

https://github.com/ampproject/amphtml/blob/3a86226fe428ce72adb67cffe2dd2f1fae278a35/extensions/amp-bodymovin-animation/validator-amp-bodymovin-animation.protoascii#L32-L37

**Comment by westonruter** [comment]
cc @ampproject/wg-caching

**Comment by honeybadgerdontcare** [comment]
@ampproject/wg-ui-and-a11y @nainar @cathyxz this has always been set as either true|false|number. Is `number` meant to be an actual numerical value? If so, what range should it be? The rules could be updated to a `value_regex_casei` instead of a `value_casei`.

**Comment by honeybadgerdontcare** [comment]
@nainar I'm unsure if a numerical value is supported by the JavaScript. Could you take a look? It could be that `number` should be removed from validation and documentation or get supported by the code. If it's a positive integer then we can switch to a `value_regex` with `[0-9]+|false|true`.

**Comment by westonruter** [comment]
According to the [component docs](https://amp.dev/documentation/components/amp-bodymovin-animation/), the `loop` attribute: 

> Indicates whether the animation should be looping or not. By default, this attribute is set to `true`. Values for this attribute can be: `true`, `false`, or a number value. If a number is specified, the animation loops that number of times.

Numerical values do work for this component, as can be seen by supplying `loop=3` in the [AMP Playground](https://playground.amp.dev/#share=PCFkb2N0eXBlIGh0bWw+CjxodG1sIOKaoT4KPGhlYWQ+CiAgPG1ldGEgY2hhcnNldD0idXRmLTgiPgogIDx0aXRsZT5NeSBBTVAgUGFnZTwvdGl0bGU+CiAgPGxpbmsgcmVsPSJjYW5vbmljYWwiIGhyZWY9InNlbGYuaHRtbCIgLz4KICA8bWV0YSBuYW1lPSJ2aWV3cG9ydCIgY29udGVudD0id2lkdGg9ZGV2aWNlLXdpZHRoLG1pbmltdW0tc2NhbGU9MSxpbml0aWFsLXNjYWxlPTEiPgogIDxzdHlsZSBhbXAtYm9pbGVycGxhdGU+Ym9keXstd2Via2l0LWFuaW1hdGlvbjotYW1wLXN0YXJ0IDhzIHN0ZXBzKDEsZW5kKSAwcyAxIG5vcm1hbCBib3RoOy1tb3otYW5pbWF0aW9uOi1hbXAtc3RhcnQgOHMgc3RlcHMoMSxlbmQpIDBzIDEgbm9ybWFsIGJvdGg7LW1zLWFuaW1hdGlvbjotYW1wLXN0YXJ0IDhzIHN0ZXBzKDEsZW5kKSAwcyAxIG5vcm1hbCBib3RoO2FuaW1hdGlvbjotYW1wLXN0YXJ0IDhzIHN0ZXBzKDEsZW5kKSAwcyAxIG5vcm1hbCBib3RofUAtd2Via2l0LWtleWZyYW1lcyAtYW1wLXN0YXJ0e2Zyb217dmlzaWJpbGl0eTpoaWRkZW59dG97dmlzaWJpbGl0eTp2aXNpYmxlfX1ALW1vei1rZXlmcmFtZXMgLWFtcC1zdGFydHtmcm9te3Zpc2liaWxpdHk6aGlkZGVufXRve3Zpc2liaWxpdHk6dmlzaWJsZX19QC1tcy1rZXlmcmFtZXMgLWFtcC1zdGFydHtmcm9te3Zpc2liaWxpdHk6aGlkZGVufXRve3Zpc2liaWxpdHk6dmlzaWJsZX19QC1vLWtleWZyYW1lcyAtYW1wLXN0YXJ0e2Zyb217dmlzaWJpbGl0eTpoaWRkZW59dG97dmlzaWJpbGl0eTp2aXNpYmxlfX1Aa2V5ZnJhbWVzIC1hbXAtc3RhcnR7ZnJvbXt2aXNpYmlsaXR5OmhpZGRlbn10b3t2aXNpYmlsaXR5OnZpc2libGV9fTwvc3R5bGU+PG5vc2NyaXB0PjxzdHlsZSBhbXAtYm9pbGVycGxhdGU+Ym9keXstd2Via2l0LWFuaW1hdGlvbjpub25lOy1tb3otYW5pbWF0aW9uOm5vbmU7LW1zLWFuaW1hdGlvbjpub25lO2FuaW1hdGlvbjpub25lfTwvc3R5bGU+PC9ub3NjcmlwdD4KICA8c2NyaXB0IGFzeW5jIHNyYz0iaHR0cHM6Ly9jZG4uYW1wcHJvamVjdC5vcmcvdjAuanMiPjwvc2NyaXB0PgogIDxzY3JpcHQgYXN5bmMgY3VzdG9tLWVsZW1lbnQ9ImFtcC1ib2R5bW92aW4tYW5pbWF0aW9uIiBzcmM9Imh0dHBzOi8vY2RuLmFtcHByb2plY3Qub3JnL3YwL2FtcC1ib2R5bW92aW4tYW5pbWF0aW9uLTAuMS5qcyI+PC9zY3JpcHQ+CjwvaGVhZD4KPGJvZHk+CjxhbXAtYm9keW1vdmluLWFuaW1hdGlvbgogIGxheW91dD0iZml4ZWQiCiAgd2lkdGg9IjIwMCIKICBoZWlnaHQ9IjIwMCIKICBzcmM9Imh0dHBzOi8vbmFpbmFyLmdpdGh1Yi5pby9sb2FkZXIuanNvbiIKICBsb29wPSIzIgo+CjwvYW1wLWJvZHltb3Zpbi1hbmltYXRpb24+CjwvYm9keT4KPC9odG1sPgo=).

Interestingly, a `loop` value less than `1` behaves the same as `loop=true`.

So yes, it seems should rather be `value_regex`, and it could prevent `0` via `[1-9][0-9]*|false|true`.

See also https://github.com/airbnb/lottie-web/blob/27ac4c1430086bb6fb5aaddd80c302f4a436788f/player/js/animation/AnimationItemWorkerOverride.js#L17-L24

**Comment by honeybadgerdontcare** [comment]
@westonruter Thank you for confirming that the numeric works and that `0` does not. I'll run a regression to see how many documents may be using `number`.

**Comment by honeybadgerdontcare** [comment]
This is being fixed in cl/328336975
